addons/hr_attendance/models/hr_employee.py

```diff
@@ -11,33 +11,59 @@
         related='last_attendance_id.check_out', store=True,
-        groups="hr_attendance.group_hr_attendance_user,hr.group_hr_user")
+        groups="hr_attendance.group_hr_attendance_officer,hr.group_hr_user")
     attendance_state = fields.Selection(
         string="Attendance Status", compute='_compute_attendance_state',
         selection=[('checked_out', "Checked out"), ('checked_in', "Checked in")],
-        groups="hr_attendance.group_hr_attendance_kiosk,hr_attendance.group_hr_attendance,hr.group_hr_user")
+        groups="hr_attendance.group_hr_attendance_officer,hr.group_hr_user")
     hours_last_month = fields.Float(
-        compute='_compute_hours_last_month', groups="hr_attendance.group_hr_attendance_user,hr.group_hr_user")
+        compute='_compute_hours_last_month', groups="hr_attendance.group_hr_attendance_officer,hr.group_hr_user")
     hours_today = fields.Float(
         compute='_compute_hours_today',
-        groups="hr_attendance.group_hr_attendance_kiosk,hr_attendance.group_hr_attendance,hr.group_hr_user")
+        groups="hr_attendance.group_hr_attendance_officer,hr.group_hr_user")
+    hours_previously_today = fields.Float(
+        compute='_compute_hours_today',
+        groups="hr_attendance.group_hr_attendance_officer,hr.group_hr_user")
+    last_attendance_worked_hours = fields.Float(
+        compute='_compute_hours_today',
+        groups="hr_attendance.group_hr_attendance_officer,hr.group_hr_user")
     hours_last_month_display = fields.Char(
-        compute='_compute_hours_last_month', groups="hr_attendance.group_hr_attendance_user,hr.group_hr_user")
+        compute='_compute_hours_last_month', groups="hr_attendance.group_hr_attendance_officer,hr.group_hr_user")
     overtime_ids = fields.One2many(
-        'hr.attendance.overtime', 'employee_id', groups="hr_attendance.group_hr_attendance_user,hr.group_hr_user")
+        'hr.attendance.overtime', 'employee_id', groups="hr_attendance.group_hr_attendance_officer,hr.group_hr_user")
     total_overtime = fields.Float(
         compute='_compute_total_overtime', compute_sudo=True,
-        groups="hr_attendance.group_hr_attendance_kiosk,hr_attendance.group_hr_attendance,hr.group_hr_user")
+        groups="hr_attendance.group_hr_attendance_officer,hr.group_hr_user")
+
+    def write(self, values):
+        old_officers = self.env['res.users']
+        if 'attendance_manager_id' in values:
+            old_officers = self.mapped('attendance_manager_id')
+            # Officer was added
+            if values['attendance_manager_id']:
+                officers = self.env['res.users'].browse(values['attendance_manager_id'])
+                officers_group = self.env.ref('hr_attendance.group_hr_attendance_officer', raise_if_not_found=False)
+                if officers_group and not officers.has_group('hr_attendance.group_hr_attendance_officer'):
+                    officers.sudo().write({'groups_id': [(4, officers_group.id)]})
```

Comment:
Maybe I missed, but we also need to do it on create as well... so, maybe we can make a function out of it? :thinking:  `_create_attendance_officers()` and call it from create and on write?